xdsl/dialects/stim/stim_printer_parser.py

```diff
@@ -1,25 +1,20 @@
-import abc
-from collections.abc import Callable, Iterable
-from contextlib import contextmanager
-from dataclasses import dataclass, field
-from typing import Any, TypeVar, cast
-
-from xdsl.dialects.builtin import ArrayAttr, FloatData, IntAttr
-from xdsl.ir import Attribute, Operation
-
 """
 This file implements a printer that prints to the .stim file format.
 Full documentation can be found here: https://github.com/quantumlib/Stim/blob/main/doc/file_format_stim_circuit.md
 """
 
+import abc
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import cast
 
-@dataclass(eq=False, repr=False)
-class StimPrinter:
-    stream: Any | None = field(default=None)
+from xdsl.dialects.builtin import ArrayAttr, FloatData, IntAttr
+from xdsl.ir import Attribute, Operation
+from xdsl.utils.base_printer import BasePrinter
 
-    def print_string(self, text: str) -> None:
-        print(text, end="", file=self.stream)
 
+@dataclass(eq=False, repr=False)
+class StimPrinter(BasePrinter):
     @contextmanager
     def in_braces(self):
```

Comment:
Possibly for a future PR, but I wonder if context managers for each delimiter pair should be added to `BasePrinter`. It's always annoyed me that there is `in_angle_brackets` (at least in the parser) but not the equivalents for braces, square brackets, and parentheses.

Reply:
Hundo p, just trying to make the API changes small, so we can have a focused discussion about the API in a separate PR